test/e2e.sh

```diff
@@ -0,0 +1,98 @@
+#! /bin/bash
+
+set -u -o pipefail 
+
+# Globals
+test_ns="driver-toolkit-test"
+dtk_registry_image="image-registry.openshift-image-registry.svc:5000/openshift/driver-toolkit:latest"
+
+# Get running RHCOS version
+# TODO check RT kernel version
+find_rhcos_version() {
+    worker_node=$(kubectl get nodes --selector='node-role.kubernetes.io/worker' -ojsonpath='{.items[].metadata.name}')
+    etc_os_release=$(oc debug node/${worker_node} \
+	    --quiet \
+	    -- cat /host/etc/os-release \
+	    | grep OSTREE_VERSION)
+
+    eval $(echo ${etc_os_release})
+
+    echo ${OSTREE_VERSION}
+}
+
+# Check if driver-toolkit imagestream is available
+test_imagestream() {
+    oc get imagestream/driver-toolkit -n openshift
+    exit_on_error $? "imagestream/driver-toolkit not found"
+}
+
+# Check for latest and RHCOS tags
+test_imagestream_tags() {
+    rhcos_version="$(find_rhcos_version)"
+    echo "INFO: Node RHCOS version: ${rhcos_version}"
+
+    echo "INFO: Checking imagestream/driver-toolkit tags:"
+    oc get imagestream/driver-toolkit -n openshift -o json | jq ".spec.tags[].name"
+    
+    image_latest=$(oc get imagestream/driver-toolkit -n openshift -o json \
+    	| jq ".spec.tags[] | select(.name == \"latest\") | .from.name")
+    echo "INFO: Image from latest tag: ${image_latest}"
+        
+    image_rhcos_tag=$(oc get imagestream/driver-toolkit -n openshift -o json \
+    	| jq ".spec.tags[] | select(.name == \"${rhcos_version}\") | .from.name")
+    echo "INFO: Image from RHCOS tag: ${image_rhcos_tag}"
+
+    [[ $image_latest != "" ]]
+    exit_on_error $? "imagestream/driver-toolkit latest tag has no image"
+
+    [[ $image_rhcos_tag != "" ]]
+    exit_on_error $? "imagestream/driver-toolkit RHCOS tag has no image"
+    
+    [[ $image_rhcos_tag == $image_latest ]]
+    exit_on_error $? "imagestream/driver-toolkit ${rhcos_version} tag does not match latest tag"
+}
+
+# Check that DTK contains the right kernel version
+test_kernel_version() {
+    node_kernel=$(oc get nodes -ojsonpath='{.items[].status.nodeInfo.kernelVersion}')
+    echo "INFO: Node kernel: ${node_kernel}"
+
+    dtk_release=$(oc debug -f manifests/01-kernel-version.yaml \
+                  --quiet \
+                  -n ${test_ns}\
+                  --image=${dtk_registry_image} \
```

Comment:
no need for this line, it's already in `manifests/01-kernel-version.yaml`. Namespace could be moved there as well?

Reply:
That is what I would have thought as well... but unfortunately it doesn't seem to be true. oc debug ignores the image and namespace set in the yaml... With that in mind, there is no point to have any manifests then... or even a test namespace.  We may want to add a `manifests` directory if manifests are needed for future tests.

For now, I replaced these lines with the --image-stream option to oc debug, allowing me to cleanly get rid of the last remaining global variable "dtk_registry_image".

Reply:
hum, I was confused that that ^^^, and it turns out that ... `oc debug -f` is confusing!

with `oc debug -f <your file>`, I get:
```
Pod: image-debug
quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:20f1dfae580c5fe0fde3fe284a12bc6ed6407c73f897c3131f49fa7c2f8de916
```
which is neither the right base name (`image`) nor the right image ....
whereas with `oc debug`:
```
$ oc create -f  <your file>
pod/rhcos-version created
$ oc debug pod/rhcos-version                                                     08:51:22 
Starting pod/rhcos-version-debug, command was: cat /etc/driver-toolkit-release.json
Pod IP: 10.128.2.73
If you don't see a command prompt, try pressing enter.
...
```
and
```
Pod: rhcos-version
registry.access.redhat.com/ubi8/ubi-minimal:8.5-204

Pod: rhcos-version-debug
registry.access.redhat.com/ubi8/ubi-minimal:8.5-204
```
which is consistent this time

--> forget about `oc debug -f <file>` without `--image`!